### DISCUSSION TITLE
Following clean architecture?

### DISCUSSION THREAD

**Discussion by PabloProta** [android/nowinandroid#661] [General]
So since I've started study Clean architecture I got front with some questions.   One of them is about the Domain layer, that shouldn't has any android dependency.  
![image](https://user-images.githubusercontent.com/58473708/232078519-1376110e-6857-439a-8a1a-a62aaae8060b.png)

Considering this fact, in this case could I have this module as android library?  I mean, not make sense just be a module that contains only Kotlin reference, independent of the android framework? such as `libs.kotlinx.coroutines.android`? but we depend of this "android" dependency, how clean architecture treats it?

**Comment by Moes81** [comment]
I would argue, that the fact that an Android lib module is used, instead of a plain Kotlin module, is no violation against clean architecture by itself. It's just a different type of module that comes with more possibilities, overhead and responsibilities.

**Comment by SimonMarquis** [comment]
Unfortunately, module dependencies in this project are a bit messy.
An issue has been created #707 regarding feature-to-feature dependency (and PR is pending).
And this is very similar to that.
Currently, `:core:domain` itself could be a pure JVM module, except that it has android modules as dependencies:
- `:core:data`: only some repositories and data classes are actually needed. The only reason for this module to be "android" is the `ConnectivityManagerNetworkMonitor` that should probably move elsewhere. (and a `android.util.Log` that should probably be removed)
- `:core:testing`: because of `NiaTestRunner` and `GrantPostNotificationsPermissionRule` that could be extracted in a dedicated "android" testing module.
